app/plan.go

```diff
@@ -51,7 +53,32 @@ func (s *planService) FindByName(name string) (*appTypes.Plan, error) {
 
 // DefaultPlan implements DefaultPlan method of PlanService interface
 func (s *planService) DefaultPlan() (*appTypes.Plan, error) {
-	return s.storage.FindDefault()
+	plan, err := s.storage.FindDefault()
```

Comment:
To reduce the indentation levels and avoid shadowing the err variable inside the if below I think we could try exiting earlier, as soon as we find some conditions match leaving the logic always on the first level. I was thinking in something like:

```
	plan, err := s.storage.FindDefault()
	if err == nil || err != appTypes.ErrPlanDefaultNotFound {
		return plan, err
	}
	plans, err := s.storage.FindAll()
	if err != nil {
		return nil, err
	}
	if len(plans) != 0 {
		return nil, appTypes.ErrPlanDefaultNotFound
	}
	... create plan ... 
```

Reply:
@cezarsa PTAL